.github/workflows/codeql-analysis.yml

```diff
@@ -9,6 +9,9 @@ on:
   schedule:
     - cron: '24 20 * * 3'
 
+permissions:
+  checks: write
```

Comment:
Why do these need write permissions?

Reply:
IIUC they need to be able to `POST` on `checks`:

https://docs.github.com/en/rest/reference/permissions-required-for-github-apps#permission-on-checks
https://docs.github.com/en/rest/reference/checks#update-a-check-run

Reply:
actually, i found https://github.com/github/codeql-action/issues/464 which doesn't like "checks: write".  I updated the list to include what they had, and also removed "checks" from "tests", and added `contents: read` since I didn't realize read-only access to things like that would go away.